src/lib.rs

```diff
@@ -63,13 +62,35 @@ pub mod pylib {
     }
 
     #[pyfunction]
-    pub fn refresh_logger() {
-        LOG_RESET_HANDLE.reset();
+    pub fn refresh_logger(py: Python) -> PyResult<()> {
+        use log::LevelFilter;
+        let logging = py.import("logging")?;
+        let python_log_level = logging
+            .getattr("getLogger")?
+            .call0()?
+            .getattr("level")?
+            .extract::<usize>()
+            .unwrap_or(0);
+
+        // https://docs.python.org/3/library/logging.html#logging-levels
+        let level_filter = match python_log_level {
+            0 => LevelFilter::Off,
+            1..=5 => LevelFilter::Trace,
+            6..=10 => LevelFilter::Debug,
+            11..=20 => LevelFilter::Info,
+            21..=30 => LevelFilter::Warn,
+            _ => LevelFilter::Error,
+        };
+
+        let logger = pyo3_log::Logger::default().filter(level_filter);
+        let handle = logger.install().unwrap();
+        LOG_RESET_HANDLE.get_or_init(|| handle);
```

Comment:
You should be able to put everything above in a function and then call that in the `get_or_init`
```
let func = || {
        use log::LevelFilter;
        let logging = py.import("logging")?;
        let python_log_level = logging
        ... 

        logger.install().unwrap()
}

LOG_RESET_HANDLE.get_or_init(func).reset();
```

Reply:
this will ensure that we only init once and that it's reset every time it's called

Reply:
also let's add a test for that